apps/web/src/components/createFormTemplate/createFormTemplateEditor/DraggableText.tsx

```diff
@@ -0,0 +1,57 @@
+import { Rnd, RndResizeCallback } from 'react-rnd';
+import { DraggableEventHandler } from 'react-draggable';
+import { TextFieldPosition } from './FormEditor';
+import { FaTimes } from 'react-icons/fa';
+
+export default function DraggableText({
+  onStop,
+  onResizeStop,
+  color,
+  onRemove,
+  currentPosition,
+}: {
+  onStop: DraggableEventHandler;
+  onResizeStop: RndResizeCallback;
+  initialText: string | null;
+  color: string;
+  onRemove: () => void;
+  currentPosition: TextFieldPosition;
+}) {
+  return (
+    <Rnd
```

Comment:
This way we can force the use of only one `DraggableText` at a time, which can also simplify state. We can store state of the single draggable object and then once the user confirms the position, we store that in another state we change less often.

Reply:
This would cause way less rerendering of a bunch of `DraggableText` components.

Reply:
If this is too much work we can just make this another ticket.

Reply:
I'll look into this tomorrow, I think that makes sense. I thought that the draggable component would only rerender after onDragStop or onDrag but I might be wrong. maybe its lagging because of the currentPosition stuff.